### ISSUE TITLE
"nuvla.api.api.NuvlaError: credentials required" when using insecure=False on http endpoint

### ISSUE THREAD

**Issue by konstan** [nuvla/python-library#8]
Authentication succeeds, but successive calls fail with `nuvla.api.api.NuvlaError: credentials required`.

```
/app # python
Python 3.7.3 (default, May 11 2019, 02:00:41)
[GCC 8.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from nuvla.api import Api
>>> a = Api(endpoint='http://10.0.3.50:8200', insecure=False)
>>> a.login_password('super', 'XXX')
<Response [201]>
>>> a.search('deployment', filter="state='STARTED'", select='id').resources
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/nuvla/api/api.py", line 328, in _cimi_request
    response.raise_for_status()
  File "/usr/local/lib/python3.7/site-packages/requests/models.py", line 940, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 403 Client Error: Forbidden for url: http://10.0.3.50:8200/api/deployment

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.7/site-packages/nuvla/api/api.py", line 456, in search
    resp_json = self._cimi_put(resource_type=resource_type, data=kwargs)
  File "/usr/local/lib/python3.7/site-packages/nuvla/api/api.py", line 355, in _cimi_put
    return self._cimi_request('PUT', uri, params=params, json=json, data=data)
  File "/usr/local/lib/python3.7/site-packages/nuvla/api/api.py", line 341, in _cimi_request
    raise NuvlaError(message, response)
nuvla.api.api.NuvlaError: credentials required
>>> a = Api(endpoint='http://10.0.3.50:8200', insecure=True)
>>> a.login_password('super', 'XXX')
<Response [201]>
>>> a.search('deployment', filter="state='STARTED'", select='id').resources
[]
>>>
```

**Comment by 0xbase12** [comment]
Try with insecure=True since it's a http connection `a = Api(endpoint='http://10.0.3.50:8200', insecure=False)`

**Comment by konstan** [comment]
Yes, I did. You can see this at the end.

The main question is why it succeeds to authenticate, but then complaints about missing creds.

**Comment by 0xbase12** [comment]
Resolved 0a5ee29f5c94e7bb91169543311695930e3a3ed0
I updated the documentation of python library to make it more clear.
